File: selectionSort.cpp

#include <bits/stdc++.h>

using namespace std;

void selectionSort(int arr [], int n) {
    //As it takes n-1 steps to sort an array.
    for(int i=0; i<= n-2; i++) {
        //select min
        int min = arr[i];
        int minIdx= i;
        for(int j=i+1; j<n; j++) {
            if(arr[j] < min) {
                min = arr[j];
                minIdx = j;
            }
        } 
        //swap
        std::swap(arr[i], arr[minIdx]);
    }
}

void print(int arr[] , int n) {
    for(int i=0; i<n; i++) {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

int main() {
    int arr [] = {13, 46, 24, 52, 20, 9};
    int n= 6;
    selectionSort(arr, n);
    print(arr, n);
}

File: SelectionSort.java
public class SelectionSort {

    public static void selectionSort(int arr[]) {
        // u represents first unsorted index.
        for (int u = 0; u < arr.length; u++) {

            int minIdx = u;
            for (int i = u; i < arr.length; i++) {
                if (arr[i] < arr[minIdx]) {
                    minIdx = i;
                }
            }
            // swap from minimum
            int temp = arr[u];
            arr[u] = arr[minIdx];
            arr[minIdx] = temp;
        }
    }

    public static void main(String[] args) {
        int arr[] = { 13, 46, 24, 52, 20, 9 };
        selectionSort(arr);
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
